plugins/qmlsupport/qmllistpropertyadaptor.cpp

```diff
@@ -60,9 +62,13 @@ PropertyData QmlListPropertyAdaptor::propertyData(int index) const
     auto var = object().variant(); // we need to keep that alive for the runtime of this method
     QQmlListProperty<QObject> *prop
         = reinterpret_cast<QQmlListProperty<QObject> *>(const_cast<void *>(var.data()));
+    if (!prop->count)
+        return pd;
     if (!prop || index >= prop->count(prop))
         return pd;
 
+    if (!prop->at)
```

Comment:
Doing this first (after checking for `prod != nullptr`) makes this code path faster. `prop->at` is just a pointer comparison, while the check above may call a (virtual) function.